platform/testing/end-to-end/vsp-cypress-resources.md

```diff
@@ -0,0 +1,835 @@
+# Cypress Resources Overview & Guide
+
+This guide is a comprehensive guide of Cypress related resources currently active within VSP (as of 11/2020).
+
+The following helpers and mocks are currently used within Cypress tests on `vets-website`. In addition to their directories, the purpose, an import method, and a reference of the function is supplied.
+
+# Table of Contents
+
+1. [Cypress & VAOS Helpers](#cypress-vaos-helpers)
+   1. [initApplicationMock](#init-application-mock)
+   2. [hasFocusableCount](#has-focusable-count)
+   3. [hasTabbableCount](#has-tabbable-count)
+   4. [initMockProfile](#init-mock-profile)
+   5. [initAppointmentListMock](#init-appointment-list-mock)
+   6. [createPastVAAppointments](#create-past-va-appointments)
+   7. [initCommunityCareMock](#init-community-care-mock)
+   8. [initExpressCareMocks](#init-express-care-mocks)
+   9. [initVAAppointmentMock](#init-va-appointment-mock)
+2. [Appointment Helpers](#appointment-helpers)
+   1. [chooseTypeOfCareTest](#choose-type-of-care-test)
+   2. [chooseVAFacilityTest](#choose-va-facility-test)
+   3. [chooseClinicTest](#choose-clinic-test)
+   4. [choosePreferredDateTest](#choose-preferred-date-test)
+   5. [selectTimeSlotTest](#select-time-slot-test)
+   6. [reasonForAppointmentTest](#reason-for-appointment-test)
+   7. [contactInfoTest](#contact-info-test)
+   8. [reviewTest](#review-test)
+   9. [confirmationPageTest](#confirmation-page-test)
+3. [Mocks](#mocks)
+   1. [confirmedVA](#confirmed-va)
+   2. [confirmedCC](#confirmed-cc)
+   3. [requests](#requests)
+   4. [cancelResons](#cancel-reasons)
+   5. [supportedSites](#supported-sites)
+   6. [facilities](#facilities)
+   7. [facilities983](#facilities-983)
+   8. [clinicList983](#clinic-list-983)
+   9. [slots](#slots)
+   10. [getVAAppointmentMock](#get-va-appointment-mock)
+   11. [getExpressCareRequestCriteriaMock](#get-express-care-request-criteria-mock)
+   12. [getParentSiteMock](#get-parent-site-mock)
+
+# Cypress & VAOS Helperss <a name="cypress-vaos-helpers"></a>
+
+Custom helpers are reusable components for commonly needed test situations, like loading custom data, profiles, or appointments. Below is a list of helpers currently written along with import methods, purpose, and a reference guide.
+
+Consider writing a reusable helper when an action is needed to perform certain tests. They are located in `src/applications/gi/tests/e2e/cypress-helpers.js` and `src/applications/vaos/tests/e2e/vaos-cypress-helpers.js`.
```

Comment:
These helpers are within the product folders for which the helpers are used. I'm not sure what the implication is for the scope of these helpers. Unless this is documented somewhere by Cypress that I can't find, I think we will need to experiment to see if Cypress loads these into the runner and permits their use throughout the testing suite.

At any rate, I think our recommendation should be slightly different than to encourage other product teams to be modifying the helper files located in a different team's product folder. Should each VFS team be creating a helper file in their product folder? Should we be consolidating these helper files into a larger file closer to the root?

Reply:
Solid points to consider. Seems when the tests were more localized to a particular team, it made sense to store them in that directory. Now that we're publicizing their contacts (and reusability) there's a strong argument for these to live in a parent directory. The caveat to that would be tests that are scoped and only usable within a particular directory.

Whatever way we go, DRY principles ought to be encouraged. Let's avoid having the same commonly used helper methods in various directories throughout the app.